src/python/pants/backend/codegen/protobuf/python/rules.py

```diff
@@ -99,12 +99,17 @@ async def generate_python_from_protobuf(
     protoc_gen_mypy_script = "protoc-gen-mypy"
     protoc_gen_mypy_grpc_script = "protoc-gen-mypy_grpc"
     mypy_pex = None
+    complete_pex_env = pex_environment.in_sandbox(working_directory=None)
```

Comment:
A drive-by fix?

Reply:
No, below (as you may have seen) we now need to pass `complete_pex_env` into `VenvPexRequest`. We could default it, but I like the explicitness of forcing callers to choose.